planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/path.cpp

```diff
@@ -0,0 +1,292 @@
+}
+
+ShiftLine get_lane_changing_shift_line(
+  const Pose & lane_changing_start_pose, const Pose & lane_changing_end_pose,
+  const PathWithLaneId & reference_path, const double shift_length)
+{
+  ShiftLine shift_line;
+  shift_line.end_shift_length = shift_length;
+  shift_line.start = lane_changing_start_pose;
+  shift_line.end = lane_changing_end_pose;
+  shift_line.start_idx = autoware::motion_utils::findNearestIndex(
+    reference_path.points, lane_changing_start_pose.position);
+  shift_line.end_idx = autoware::motion_utils::findNearestIndex(
+    reference_path.points, lane_changing_end_pose.position);
+
+  return shift_line;
+}
+
+ShiftedPath get_shifted_path(
+  const PathWithLaneId & target_lane_reference_path, const LaneChangeInfo & lane_change_info)
+{
+  const auto longitudinal_acceleration = lane_change_info.longitudinal_acceleration;
+
+  PathShifter path_shifter;
+  path_shifter.setPath(target_lane_reference_path);
+  path_shifter.addShiftLine(lane_change_info.shift_line);
+  path_shifter.setLongitudinalAcceleration(longitudinal_acceleration.lane_changing);
+  const auto initial_lane_changing_velocity = lane_change_info.velocity.lane_changing;
+  path_shifter.setVelocity(initial_lane_changing_velocity);
+  path_shifter.setLateralAccelerationLimit(std::abs(lane_change_info.lateral_acceleration));
+
+  constexpr auto offset_back = false;
+  ShiftedPath shifted_path;
+  [[maybe_unused]] const auto success = path_shifter.generate(&shifted_path, offset_back);
+  return shifted_path;
+}
+
+std::optional<double> exceed_yaw_threshold(
+  const PathWithLaneId & prepare_segment, const PathWithLaneId & lane_changing_segment,
+  const double yaw_th_rad)
+{
+  const auto & prepare = prepare_segment.points;
+  const auto & lane_changing = lane_changing_segment.points;
+
+  if (prepare.size() <= 1 || lane_changing.size() <= 1) {
+    return std::nullopt;
+  }
+
+  const auto & p1 = std::prev(prepare.end() - 1)->point.pose;
+  const auto & p2 = std::next(lane_changing.begin())->point.pose;
+  const auto yaw_diff_rad = std::abs(autoware::universe_utils::normalizeRadian(
+    tf2::getYaw(p1.orientation) - tf2::getYaw(p2.orientation)));
+  if (yaw_diff_rad > yaw_th_rad) {
+    return yaw_diff_rad;
+  }
+  return std::nullopt;
+}
+};  // namespace
+
+namespace autoware::behavior_path_planner::utils::lane_change
+{
+using behavior_path_planner::lane_change::CommonDataPtr;
+
+bool get_prepare_segment(
+  const CommonDataPtr & common_data_ptr, const PathWithLaneId & prev_module_path,
+  const LaneChangePhaseMetrics prep_metric, PathWithLaneId & prepare_segment)
+{
+  const auto & current_lanes = common_data_ptr->lanes_ptr->current;
+  const auto & target_lanes = common_data_ptr->lanes_ptr->target;
+  const auto backward_path_length = common_data_ptr->bpp_param_ptr->backward_path_length;
+
+  if (current_lanes.empty() || target_lanes.empty()) {
+    throw std::logic_error("lane change start is behind target lanelet!");
```

Comment:
Is this message correct ?

Reply:
Fixed the mistake in https://github.com/autowarefoundation/autoware.universe/pull/9633/commits/8ecae27965523167783b645ffb4a543aab783a35